Retain not in-scope URLs

adeotek opened the pull request:
Config option "keepNotInScopeLinks" added.
If set `true`, the links that are not "in-scope" will be stored in `collector.not-in-scope-urls`.

essiembre commented:
Thanks for your pull-request.  In addition to adding the not-in-scope URLs to the metadata, you are also adding it to the "crawl data" (HttpCrawlData).  The crawl data is meant to cache information that is needed by the crawl process (current or subsequent).  The crawler itself has no use for not-in-scope URLs so we should only store it as metadata (for submission to the committer).   Do you see a reason for having those in the crawl "store"?  If not, I would remove it from HttpCrawlData to keep the crawl "store" as compact as we can.

adeotek commented:
You are right, there is no need for this information in HttpCrawlData, so I've removed it.
Also I have corrected a method name in HttpMetadata (getDocumentRejectedUrls changed to getDocumentNotInScopeUrls).

Please let me know it there are other issues.

essiembre commented:
I made minor renames and also ensured out of scope URLs are not queued.   Thanks again for your contribution!
